samples/SampleApp/Program.cs

```diff
@@ -22,15 +22,17 @@ public static void Main(string[] args)
             CustomRouter();
 
             CustomApplicationBuilder();
+            
+            CustomMiddlewareDelegate();
 
             StartupClass(args);
         }
 
         private static void HelloWorld()
         {
-            using (WebHost.Start(context => context.Response.WriteAsync("Hello, World!")))
+            using (var host = WebHost.Start(context => context.Response.WriteAsync("Hello, World!")))
             {
-                //host.WaitForShutdown(); // TODO: https://github.com/aspnet/Hosting/issues/1022
+                host.WaitForShutdown();
                 Console.WriteLine("Running HelloWorld: Press any key to shutdown and start the next sample...");
```

Comment:
This needs more cleanup, it prevents the Console.WriteLine, the message is outdated, and is redundant with the ReadKey